spec/application_integration_spec.rb

```diff
@@ -0,0 +1,135 @@
+require 'database_connection'
+require 'item_repository'
+require 'order_repository'
+require_relative '../app'
+
+RSpec.describe Application do
+  # snippet and function call copied and modified from repository class recipe
+  def reset_tables
+    seed_sql = File.read('spec/seeds_shop_manager.sql')
+    connection = PG.connect({ host: '127.0.0.1', dbname: 'shop_manager_test' })
+    connection.exec(seed_sql)
+  end
+
+  before(:each) do 
+    reset_tables
+  end
+
+  describe "#list_all_items" do
+    it "prints all items" do
+      database_name = 'shop_manager_test'
+      terminal = double(:terminal)
+      item_repository = ItemRepository.new
+      order_repository = OrderRepository.new
+      app = Application.new(database_name, terminal,
+        item_repository, order_repository)
+      menu_string = "Welcome to the shop management program!\n\n" \
+        "What do you want to do?\n" \
+        "  1 = list all shop items\n" \
+        "  2 = create a new item\n" \
+        "  3 = list all orders\n" \
+        "  4 = create a new order"
+      expect(terminal).to receive(:puts).with(menu_string).ordered
+      expect(terminal).to receive(:gets).and_return('1').ordered
```

Comment:
So, if you are able to organise it with those tools I mentioned, each it block can start with the `gets` of which menu item has been selected! Other than that your tests are very clean and tight!